loader/loader.c

```diff
@@ -3005,13 +3001,17 @@ VkResult add_data_files(const struct loader_instance *inst, char *search_path, s
                 break;
             }
         } else {  // Otherwise, treat it as a directory
-            dir_stream = loader_opendir(inst, cur_file);
+            DIR *dir_stream = loader_opendir(inst, cur_file);
             if (NULL == dir_stream) {
                 continue;
             }
             while (1) {
-                dir_entry = readdir(dir_stream);
-                if (NULL == dir_entry) {
+                errno = 0;
+                struct dirent *dir_entry = readdir(dir_stream);
+                if (errno != 0) {
+                    loader_log(inst, VULKAN_LOADER_ERROR_BIT, 0,
+                               "readdir failed to read the next file in the directory with errno set to %d", errno);
```

Comment:
`...%s", strerror(errno)` maybe? At least that way you get a readable-ish description maybe sometimes.

Reply:
Good idea - isn't bad to try to get a real error message from the OS in the first place.